### ISSUE TITLE
TensorFlow2.1.0 ‘SavedModel’ Format for FEMNIST dataset

### ISSUE THREAD

**Issue by jingoohan** [IBM/federated-learning-lib#79]
FedProx of IBM FL requires to use TensorFlow2.1.0 ‘SavedModel’ Format.
I tried to test the following CNN model for FEMNIST dataset as the LEAF framework used. 
<img width="492" alt="Screen Shot 2021-08-26 at 7 02 00 PM" src="https://user-images.githubusercontent.com/41449519/131047203-2ebbed07-4d6a-4b89-af40-fb743df63117.png">

But, it makes errors as below.
<img width="1153" alt="Screen Shot 2021-08-26 at 7 11 26 PM" src="https://user-images.githubusercontent.com/41449519/131047553-4ac0edc3-bfd6-4ee9-ad58-0b6872a9b267.png">

Would you share CNN model of TensorFlow2.1.0 ‘SavedModel’ Format for FEMNIST dataset?
Or is it possible to use the following TensorFlow2.1.0 ‘HDF5’ Format for FedProx?
I checked the current code, but it does not look that FedProx can support  ‘HDF5’ Format .
<img width="490" alt="Screen Shot 2021-08-26 at 7 13 13 PM" src="https://user-images.githubusercontent.com/41449519/131047654-11a8f8a8-7059-47b7-bf73-4c1a04f34610.png">

**Comment by Yi-Zoey** [comment]
Hi can you share the code you use to define the model? I think you should also follow the [example](https://github.com/IBM/federated-learning-lib/blob/main/examples/fedprox/model.py) where it also defines the config.

**Comment by jingoohan** [comment]
Hi @Yi-Zoey 

I share my model code (examples/fedprox/model.py). This CNN model is as same as the LEAF framework used for training FEMNIST dataset.

Unlike TFv1.0 HDF5 format, it is very tricky to make more complex CNN model using TFv2.1.0 SavedModel format. It makes a lot of runtime errors whenever I change the default simple CNN model that IBM FL already provided. 

------------------------------------------------------------
import copy

import tensorflow as tf
from tensorflow.keras import Model
from tensorflow.keras.layers import Dense, Flatten, Conv2D, MaxPooling2D

class MyModel(Model):
    def __init__(self):
        super(MyModel, self).__init__()
        self.conv1 = Conv2D(32, 5, activation='relu', padding='same', input_shape=(28, 28, 1))
        self.conv2 = Conv2D(64, 5, activation='relu', padding='same')
        self.maxpool = MaxPooling2D(2, strides=2, padding='same')
        self.flatten = Flatten()
        self.d1 = Dense(2048, activation='relu')
        self.d2 = Dense(62, activation='softmax')

    def call(self, x):
        x = self.conv1(x)
        x = self.maxpool(x)
        x = self.conv2(x)
        x = self.maxpool(x)
        x = self.flatten(x)
        x = self.d1(x)
        return self.d2(x)

    def get_config(self):
        layer_configs = []
        for layer in super(MyModel, self).layers:
            layer_configs.append(tf.keras.utils.serialize_keras_object(layer))

        config = {
            'name': self.name,
            'layers': copy.deepcopy(layer_configs)
        }

        return config

    @classmethod
    def from_config(cls, config):

        layer_configs = config['layers'] if 'name' in config else config
        model = cls()
        for i, layer in enumerate(model.layers):
            layer.from_config(layer_configs[i]['config'])

        return model
------------------------------------------------------------
Thank you.

**Comment by Yi-Zoey** [comment]
@jingoohan I think in the FedProx Example, it doesn't save the compiled model. It saves the architecture and compiles it when parties start loading the model config, please check the code from [here](https://github.com/IBM/federated-learning-lib/blob/main/examples/fedprox/generate_configs.py#L64).

**Comment by jingoohan** [comment]
Hi @Yi-Zoey 

There has been some misunderstanding. Unlike other example codes, current code for FedProx uses separate codes: generate_configs.py and model.py. So, generate_configs.py calls CNN model in model.py. I also checked that model_architecture.json is created. 

As you mentioned, party loads the model config, saves and compile the model. But, it makes runtime errors when I try to train the above CNN model. On the contrary, in case of training the default simple CNN model on MNIST dataset, it does not make any runtime issues.

You guys can simply check this issue after changing the default CNN model in examples/fedprox/model.py.

**Comment by jingoohan** [comment]
Hi @Yi-Zoey 

I resolved this issue modifying femnist_keras_data_handler.py. 
Thank you.
